pkg/cmd/cli/sa/newtoken.go

```diff
@@ -0,0 +1,208 @@
+package sa
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/serviceaccounts"
+	osautil "github.com/openshift/origin/pkg/serviceaccounts/util"
+)
+
+const (
+	NewServiceAccountTokenRecommendedName = "new-token"
+
+	newServiceAccountTokenShort = `Generate a new service account API token for a service account.`
+
+	newServiceAccountTokenLong = `
+Generate a new service account API token for a service account.
+
+Service account API tokens are used by service accounts for privileged access to the API. This command will generate a new
```

Comment:
Service account API tokens are used to authenticate to the API server, not for privileged access.  Their access isn't privileged.